typed-racket-lib/typed-racket/utils/mutated-vars.rkt

```diff
@@ -17,8 +19,18 @@
         (loop stx tbl)))
     (syntax-parse stx
       #:literal-sets (kernel-literals)
+      ;; let us care about custom syntax classes
+      [form
```

Comment:
Why do we need this additional case? It seemed from the earlier discussion like skipping some `set!` cases would be all that's needed here.